sqlmesh/core/engine_adapter/athena.py

```diff
@@ -0,0 +1,441 @@
+                ):
+                    column_def = exp.ColumnDef(this=exp.to_identifier(match_name), kind=match_dtype)
+                    schema_expressions.append(column_def)
+            else:
+                schema_expressions = partitioned_by
+
+            properties.append(
+                exp.PartitionedByProperty(this=exp.Schema(expressions=schema_expressions))
+            )
+
+        if clustered_by:
+            # Athena itself supports CLUSTERED BY, via the syntax CLUSTERED BY (col) INTO <n> BUCKETS
+            # However, SQLMesh is more closely aligned with BigQuery's notion of clustering and
+            # defines `clustered_by` as a List[str] with no way of indicating the number of buckets
+            #
+            # Athena's concept of CLUSTER BY is more like Iceberg's `bucket(<num_buckets>, col)` partition transform
+            logging.warning("clustered_by is not supported in the Athena adapter at this time")
+
+        if storage_format:
+            if is_iceberg:
+                # TBLPROPERTIES('format'='parquet')
+                table_properties["format"] = exp.Literal.string(storage_format)
+            else:
+                # STORED AS PARQUET
+                properties.append(exp.FileFormatProperty(this=storage_format))
+
+        if table and (location := self._table_location(table_properties, table)):
+            properties.append(location)
+
+        for name, value in table_properties.items():
+            properties.append(exp.Property(this=exp.var(name), value=value))
+
+        if is_iceberg and expression:
+            # To make a CTAS expression persist as iceberg, alongside setting `table_type=iceberg` (which the user has already
+            # supplied in physical_properties and is thus set above), you also need to set:
+            #  - is_external=false
+            #  - table_location='s3://<path>'
+            # ref: https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html#ctas-table-properties
+            properties.append(exp.Property(this=exp.var("is_external"), value="false"))
+
+        if properties:
+            return exp.Properties(expressions=properties)
+
+        return None
+
+    def _truncate_table(self, table_name: TableName) -> None:
+        table = exp.to_table(table_name)
+        # Athena doesnt support TRUNCATE TABLE. The closest thing is DELETE FROM <table> but it only works on Iceberg
+        self.execute(f"DELETE FROM {table.sql(dialect=self.dialect, identify=True)}")
+
+    def _table_type(
+        self, table_properties: t.Optional[t.Dict[str, exp.Expression]] = None
+    ) -> t.Union[t.Literal["hive"], t.Literal["iceberg"]]:
+        """
+        Use the user-specified table_properties to figure out of this is a Hive or an Iceberg table
+        """
+        # if table_type is not defined or is not set to "iceberg", this is a Hive table
+        if table_properties and (table_type := table_properties.get("table_type", None)):
```

Comment:
Why are we not using `storage_format` for this instead?

Reply:
Basically any value that is not `iceberg` is treated as `hive`.

Reply:
I thought about `storage_format` but decided not to use it because it describes a different concept. Both Hive and Iceberg tables support different storage formats.

For example, a Hive table can be `STORED AS PARQUET` or `STORED AS ORC` or if you really dont like your colleagues `STORED AS TEXTFILE`.

Same for Iceberg, the internal `storage_format` can be set to `parquet` or `orc` or whatever the engine supports.

So `storage_format=hive` / `storage_format=parquet` doesn't make sense because they're _table_ formats. We dont have a top-level `table_format` property and I didnt want to add one just for Athena

Reply:
FWIW, when used with spark `iceberg` is provided through `storage_format` because the SparkSQL syntax looks like: 
```
CREATE TABLE ... USING [iceberg|parquet|etc]
```
Should we be consistent?

Reply:
If you did that, how would you specify "Hive + Parquet", "Hive + ORC", "Iceberg + Parquet", "Iceberg + ORC" etc?

The table format is independent of the storage format.

`storage_format` is [currently defined](https://sqlmesh.readthedocs.io/en/stable/concepts/models/overview/?h=storage#storage_format) as:
```
Storage format is a property for engines such as Spark or Hive that support storage formats such as parquet and orc.
```
in the docs which I think makes sense. It mentions the actual format of the files on disk, eg Parquet or ORC. Not the type of table that is managing those files. 

The [Hive syntax](https://iceberg.apache.org/docs/latest/hive/#non-partitioned-tables) is:
```
CREATE TABLE x (i int) STORED BY ICEBERG STORED AS ORC;
```

Which differentiates `STORED BY` (table format) and `STORED AS` (storage format). `STORED AS ICEBERG` is invalid.

I'm not super familiar with Spark but it looks like `USING` [triggers a "Data Source"](https://spark.apache.org/docs/latest/sql-ref-syntax-ddl-create-table-datasource.html#parameters) and then if Iceberg is available on the classpath, you can `USING iceberg` and then [set the storage format](https://iceberg.apache.org/docs/latest/spark-ddl/#create-table) in `TBLPROPERTIES` (or in the Iceberg catalog config). So it definitely blurs these concepts compared to other engines

Reply:
I guess if we wanted to represent both concepts using the `storage_format` property, we would need to overload the syntax and do some basic parsing.

Something like:

`storage_format iceberg` = iceberg + whatever default is configured in the engine
`storage_format iceberg[parquet]` = iceberg + parquet
`storage_format iceberg[orc]` = iceberg + orc

etc

Reply:
Honestly, I haven't seen anyone using iceberg with anything other than parquet, so values `parquet`, `orc` automatically assume Hive, while `iceberg` just assumes iceberg with parquet as per CREATE TABLE syntax in spark: https://iceberg.apache.org/docs/nightly/spark-ddl/#create-table

Reply:
IMHO, I wouldn't worry too much about ORC, at least in the context of Iceberg.

Reply:
I think the bigger picture here is that table formats and storage formats are two separate concepts and we should not conflate the two.

We also shouldn't just assume that people will only ever want to use parquet and hardcode it, isnt the point of `physical_properties` to expose features of the underlying database system?

I can already imagine someone with an established data lake that chose ORC for whatever reason and has downstream consumers that expect Iceberg/ORC tables, and now they cant easily use SQLMesh because SQLMesh will only create Iceberg/Parquet tables.

We also need to add DBT support and even DBT [clearly separates these concepts](https://github.com/dbt-labs/dbt-athena/tree/main?tab=readme-ov-file#iceberg)

Reply:
@erindru @izeigerman few comments from my side about this, after maintaining dbt-athena for 2 years.

People want to be in control of the iceberg tables that they use.
What @izeigerman raised 
> I haven't seen anyone using iceberg with anything other than parquet, so values
it's correct, but as a user I want to be in control of the data format used by Iceberg, therefore they might team that want to use ORC, other that want to use parquet, but the user must be in control, and have the possibility to decide.

This is valid for also for table properties, full list is available [here](https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html#ctas-table-properties).
There are certain property that can be templated of course, but some other properties like vacuum_min_snapshots_to_keep (just to mention one), where the final user must be in control.

Reply:
Thanks Nicola, appreciate your insight!